test/gtest/uct/test_md.cc

```diff
@@ -153,11 +153,14 @@ void test_md::test_reg_mem(unsigned access_mask,
 std::vector<test_md_param> test_md::enum_mds(const std::string& cmpt_name) {
 
     std::vector<md_resource> md_resources = enum_md_resources();
+    static const char *str                = getenv("GTEST_UCT_MD_NAME");
+    std::string md_name                   = (str != NULL) ? str : "";
 
     std::vector<test_md_param> result;
     for (std::vector<md_resource>::iterator iter = md_resources.begin();
          iter != md_resources.end(); ++iter) {
-        if (iter->cmpt_attr.name == cmpt_name) {
+        if ((iter->cmpt_attr.name == cmpt_name) &&
+            ((md_name == "") || (md_name == iter->rsc_desc.md_name))) {
```

Comment:
minor: more C++ idiomatic way of checking empty string:
`md_name.empty()`

Reply:
fixed